gallery/plot_transforms_v2_e2e.py

```diff
@@ -0,0 +1,147 @@
+    # This loads fake data for illustration purposes of this example. In practice, you'll have
+    # to replace this with the proper data
+    root = pathlib.Path("assets") / "coco"
+    return datasets.CocoDetection(str(root / "images"), str(root / "instances.json"), **kwargs)
+
+
+dataset = load_example_coco_detection_dataset()
+
+sample = dataset[0]
+image, target = sample
+print(type(image))
+print(type(target))
+print(type(target[0]))
+print(list(target[0].keys()))
+
+
+########################################################################################################################
+# The dataset returns a two-tuple with the first item being a :class:`PIL.Image.Image` and second one a list of
+# dictionaries, which each containing the annotations for a single object instance. As is, this format is not compatible
+# with the transforms v2, but also the models. To overcome that, we provide the
+# :func:`~datasets.wrap_dataset_for_transforms_v2` function. For :class:`~datasets.CocoDetection` this changes the
+# target structure to a single dictionary of lists. It also adds the key-value-pairs `"boxes"`, `"masks"`, and
+# `"labels"` wrapped in the corresponding `torchvision.datapoints`.
+
+dataset = datasets.wrap_dataset_for_transforms_v2(dataset)
+
+sample = dataset[0]
+image, target = sample
+print(type(image))
+print(type(target))
+print(list(target.keys()))
+print(type(target["boxes"]))
+
+########################################################################################################################
+# As baseline, let's have a look how a sample looks like without transformations
+
+show(sample)
+
+
+########################################################################################################################
+# With the dataset properly set up, we can now define the augmentation pipeline. This is done the same way it is done in
+# transforms v1, but now handles bounding boxes without any extra configuration.
+#
+# .. note::
+#    Although the :class:`SanitizeBoundingBoxes` transform is a no-op in this example, it should be placed at least once
+#    at the end of a detection pipeline to remove degenerate bounding boxes as well as the corresponding labels and
+#    optionally masks.
+
+transform = transforms.Compose(
+    [
+        transforms.RandomPhotometricDistort(),
+        transforms.RandomZoomOut(fill=defaultdict(lambda: 0, {PIL.Image.Image: (123, 117, 104)})),
+        transforms.RandomIoUCrop(),
+        transforms.RandomHorizontalFlip(),
+        transforms.ToImageTensor(),
+        transforms.ConvertImageDtype(torch.float32),
+        # FIXME: this is not good UX
+        transforms.SanitizeBoundingBoxes(labels_getter=lambda sample: sample[1]["labels"]),
+    ]
+)
+
+########################################################################################################################
+# Let's look how the sample looks like with our augmentation pipeline in place
+# sphinx_gallery_thumbnail_number = 2
```

Comment:
> sphinx_gallery_thumbnail_number = 2

Looks like this is being rendered as pure text, consider adding it to the top as in https://github.com/pytorch/vision/blob/main/gallery/plot_optical_flow.py#L25.